adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -490,6 +523,19 @@ private AuthenticationResult postMessage(String requestMessage, Map<String, Stri
             Logger.e(TAG, e.getMessage(), "", ADALError.ENCODING_IS_NOT_SUPPORTED, e);
             throw e;
         } catch (IOException e) {
+            //retry once if there is an observation of a network timeout by the client 
+            if (e instanceof SocketTimeoutException){
```

Comment:
because its not a good programming practice. This of it this way. we replace this entire try catch block with a catch(Exception e) {
if (e instanceof ...){
} ELSE if (e instanceof ...){
}

}
and so on... just consistency demands that we catch exceptions individually. 

---
In reply to: [71401988](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/676#discussion_r71401988) [](ancestors = 71401988)